Jenkinsfile

```diff
@@ -44,61 +44,76 @@ pipeline {
       }
 
       steps {
-        sh '''
-          az storage blob download \
-          --account-name $AZURE_STORAGE_ACCOUNT \
-          --container-name $AZURE_STORAGE_CONTAINER_NAME \
-          --name "$GPG_FILENAME" \
-          --file "$GPG_FILE"
-        '''
-        stash includes: "$GPG_FILE", name: 'GPG'
+        container('azure-cli') {
+          sh '''
+            az storage blob download \
+            --account-name $AZURE_STORAGE_ACCOUNT \
+            --container-name $AZURE_STORAGE_CONTAINER_NAME \
+            --name "$GPG_FILENAME" \
+            --file "$GPG_FILE"
+          '''
+          stash includes: "$GPG_FILE", name: 'GPG'
+
+        }
       }
     }
     
     stage('Download WAR archive to package'){
       steps{
-        checkout scm
-        sh '''
-          /entrypoint
-          /packaging/utils/getJenkinsVersion.py
-        '''
-        archiveArtifacts artifacts: "*.war"
-        stash includes: "$WAR", name: "WAR"
-
+        container('packaging'){
+          checkout scm
+          sh '''
+            /entrypoint
+            /packaging/utils/getJenkinsVersion.py
+          '''
+          archiveArtifacts artifacts: "*.war"
+          stash includes: "$WAR", name: "WAR"
+        }
       }
     }
     stage('Build'){
       failFast false
       parallel {
         stage('Debian') {
           steps {
-            unstash 'GPG'
-            unstash 'WAR'
-            sh 'make deb'
-            archiveArtifacts artifacts: "target/debian/*.deb"
+            container('packaging'){
+              unstash 'GPG'
+              unstash 'WAR'
+              sh 'make deb'
+              archiveArtifacts artifacts: "target/debian/*.deb"
+            }
           }
         }
         stage('Redhat') {
           steps {
-            unstash 'GPG'
-            unstash 'WAR'
-            sh 'make rpm'
-            archiveArtifacts artifacts: "target/rpm/*.rpm"
+            container('packaging'){
+              unstash 'GPG'
+              unstash 'WAR'
+              sh 'make rpm'
+              archiveArtifacts artifacts: "target/rpm/*.rpm"
+            }
           }
         }
         stage('Suse') {
           steps {
-            unstash 'GPG'
-            unstash 'WAR'
-            sh 'make suse'
-            archiveArtifacts artifacts: "target/suse/*.rpm"
+            container('packaging'){
+              unstash 'GPG'
+              unstash 'WAR'
+              sh 'make suse'
+              archiveArtifacts artifacts: "target/suse/*.rpm"
+            }
           }
         }
         stage('Windows') {
+          agent {
+            label 'jnlp-windows'
+          }
           steps {
-            unstash 'GPG'
-            unstash 'WAR'
-            sh 'echo "Running on windows"'
+            container('jnlp-window'){
```

Comment:
Is this the actual label? `window`?

Reply:
Just a typo in a placeholder, I don't know yet how I will package windows application :D